site/app/libraries/database/DatabaseQueries.php

```diff
@@ -275,6 +281,10 @@ public function searchThreads($searchQuery){
     	return $this->course_db->rows();
     }
 
+    public function threadExists(){
```

Comment:
I am not sure about this one.
Should we consider deleted thread or not, while checking threadExists.

Rest looks good to me.